src/projectname.rs

```diff
@@ -3,21 +3,21 @@ use heck::{KebabCase, SnakeCase};
 /// Stores user inputted name and provides convenience methods
 /// for handling casing.
 pub struct ProjectName {
-    user_input: String,
+    pub user_input: String,
+    pub kebab_case_name: String,
+    pub snake_case_name: String,
 }
 
 impl ProjectName {
     pub fn new(name: &str) -> ProjectName {
         ProjectName {
             user_input: name.to_string(),
+            kebab_case_name: name.to_kebab_case(),
+            snake_case_name: name.to_snake_case(),
         }
     }
 
-    pub fn kebab_case(&self) -> String {
-        self.user_input.to_kebab_case()
-    }
```

Comment:
Super tiny nit: I'd keep these methods as getters instead of using the field names (and make all fields private), so it's easier to change this again in the future.

What you could however do is give these semantically meaningful names, like `as_crate_name()` and `as_identifier()`.

Reply:
Thanks for this review :)
The `crate_name` is kebab-case right?

Reply:
Is the nit about getters instead of using field names is part of another Rust tips than it will be easier to change? (just for my culture)

Reply:
Yeah, crate names are usually written in kebab-case (snake_case is also allowed).

Using getters is something I personally tend to do quite often, partially also because I also like to add random debug statements and breakpoints (which is very easy to add to a function but practically impossible to add to struct field accesses). The official API guidelines have a recommendation in their [future proofing](https://rust-lang-nursery.github.io/api-guidelines/future-proofing.html#structs-have-private-fields-c-struct-private) chapters to not use public fields because changing them is a breaking change to your crate's API. (This is of course less of an issue for a binary crate.)

Reply:
Great to hear!
Thanks :)